### PR TITLE
Enable native authentication in MySQL

### PR DESCRIPTION
**Pull request by Mrod1598**
**Description:** <Describe what has changed.>
Add support for native authentication in mysql.

### CODE REVIEW

File: receiver/mysqlreceiver/client.go [hunk 1]
```diff
@@ -38,11 +38,12 @@ var _ client = (*mySQLClient)(nil)
...
-		Passwd: conf.Password,
-		Net:    conf.Transport,
-		Addr:   conf.Endpoint,
-		DBName: conf.Database,
+		User:                 conf.Username,
+		Passwd:               conf.Password,
+		Net:                  conf.Transport,
+		Addr:                 conf.Endpoint,
+		DBName:               conf.Database,
+		AllowNativePasswords: true,
```
Comment (jpkrohling) [inline receiver/mysqlreceiver/client.go:46]: It seems wrong to just add this without giving users the chance to customize it. Besides, the default value for this seems to indeed be true: what would this change bring?

https://pkg.go.dev/github.com/go-sql-driver/mysql#Config

Comment (Mrod1598) [reply receiver/mysqlreceiver/client.go:46]: It seems that the default was incorrect in testing as it was building the DSN by setting this parameter to false. I'll add it as a config option but default to true.

**Review by jpkrohling — APPROVED** [review]

File: receiver/mysqlreceiver/README.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ The following settings are optional:
 - `endpoint`: (default = `localhost:3306`)
 - `username`: (default = `root`)
 - `password`: The password to the username.
+- `AllowNativePasswords`: (default = `true`)
```
Comment (djaglowski) [inline receiver/mysqlreceiver/README.md:22]:
```suggestion
- `allow_native_passwords`: (default = `true`)
```

File: receiver/mysqlreceiver/config.go [hunk 3]
```diff
@@ -24,5 +24,6 @@ type Config struct {
 	Username                                string `mapstructure:"username,omitempty"`
 	Password                                string `mapstructure:"password,omitempty"`
 	Database                                string `mapstructure:"database,omitempty"`
+	AllowNativePasswords                    bool   `mapstructure:"database,omitempty"`
```
Comment (djaglowski) [inline receiver/mysqlreceiver/config.go:27]:
```suggestion
	AllowNativePasswords                    bool   `mapstructure:"allow_native_passwords,omitempty"`
```

**Review by djaglowski — CHANGES_REQUESTED** [review]

File: receiver/mysqlreceiver/config.go (hunk 3 above)
Comment (Mrod1598) [reply receiver/mysqlreceiver/config.go]: updated

File: receiver/mysqlreceiver/README.md (hunk 2 above)
Comment (Mrod1598) [reply receiver/mysqlreceiver/README.md]: updated

**Review by djaglowski — APPROVED** [review]

File: receiver/mysqlreceiver/README.md (hunk 2 above)
Comment (jpkrohling) [reply receiver/mysqlreceiver/README.md]: Good catch!

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**